### PR TITLE
fix _backported_all_estimators import error

### LINKED ISSUE #11286: [BUG] import error _backported_all_estimators  from sklearn.utils.tests import ignore_warnings deprecated
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

- Client: 2.9.2
- Tracking server: 1.x.y

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 
- **Python version**:
- **yarn version, if running the dev UI**:

### Describe the problem

The most recent version of sklearn, has removed the sklearn.utils.testing module.  Instead, use sklearn.utils._testing instead of it
mlflow.sklearn.utils._all_estimators() make cause error

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
from sklearn import datasets
import mlflow
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

dataset = datasets.load_iris()
X_train, X_test, y_train, y_test = train_test_split(dataset.data, dataset.target, test_size=0.4)

clf = LogisticRegression()
clf.fit(X_train, y_train)

mlflow.sklearn.autolog()
```

### Stack trace

```
/home/zhouyou/miniconda3/bin/python /home/zhouyou/PycharmProjects/mlflow_test/Chapter1/run1.py 
Traceback (most recent call last):
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/sklearn/utils.py", line 858, in _all_estimators
    return all_estimators()
           ^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/sklearn/utils/discovery.py", line 70, in all_estimators
    module = import_module(module_name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/sklearn/utils/metadata_routing.py", line 12, in <module>
    from ._metadata_requests import WARN, UNUSED, UNCHANGED  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/sklearn/utils/_metadata_requests.py", line 87, in <module>
    from ..exceptions import UnsetMetadataPassedError
ImportError: cannot import name 'UnsetMetadataPassedError' from 'sklearn.exceptions' (/home/zhouyou/miniconda3/lib/python3.11/site-packages/sklearn/exceptions.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/zhouyou/PycharmProjects/mlflow_test/Chapter1/run1.py", line 12, in <module>
    mlflow.sklearn.autolog()
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/utils/autologging_utils/__init__.py", line 424, in autolog
    return _autolog(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/sklearn/__init__.py", line 1261, in autolog
    _autolog(
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/sklearn/__init__.py", line 1822, in _autolog
    estimators_to_patch = _gen_estimators_to_patch()
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/sklearn/__init__.py", line 98, in _gen_estimators_to_patch
    _, estimators_to_patch = zip(*_all_estimators())
                                  ^^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/sklearn/utils.py", line 860, in _all_estimators
    return _backported_all_estimators()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zhouyou/miniconda3/lib/python3.11/site-packages/mlflow/sklearn/utils.py", line 905, in _backported_all_estimators
    from sklearn.utils.testing import ignore_warnings
ModuleNotFoundError: No module named 'sklearn.utils.testing'

Process finished with exit code 1
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by zhouyou9505**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/zhouyou9505/mlflow/pull/11304?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11304/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11304
```

</p>
</details>

### Related Issues/PRs

Fix #11286 

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/sklearn/test_sklearn_utils.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+@patch('sklearn.utils.all_estimators', side_effect=ImportError)
+def test_sklearn_utils_backported_all_estimators(mock_all_estimators):
+    _all_estimators()
+    _testing_check = find_importer('sklearn.utils._testing', 'ignore_warnings')
+    assert _testing_check == True
+
+    testing_check = find_importer('sklearn.utils.testing', 'ignore_warnings')
+    assert testing_check == False
```
Comment (harupy) [inline tests/sklearn/test_sklearn_utils.py:20]: I'm not a big fan of testing private functions like this. Can we remove this?

File: mlflow/sklearn/utils.py [hunk 2]
```diff
@@ -918,7 +918,10 @@ def _backported_all_estimators(type_filter=None):
         RegressorMixin,
         TransformerMixin,
     )
-    from sklearn.utils.testing import ignore_warnings
+    try:
+        from sklearn.utils.testing import ignore_warnings
+    except ImportError:
+        from sklearn.utils._testing import ignore_warnings
```
Comment (harupy) [inline mlflow/sklearn/utils.py:924]:
```suggestion
    from sklearn.utils._testing import ignore_warnings
```

https://github.com/scikit-learn/scikit-learn/blob/0.23.2/sklearn/utils/__init__.py#L1146 uses `sklearn.utils._testing`

Comment (zhouyou9505) [reply mlflow/sklearn/utils.py]: ok . sure . I had removed it . please check again.

**Comment by zhouyou9505** [comment]
ok . sure . I have removed it . please check again.

**Comment by harupy** [comment]
Thanks for the update! Triggered CI.

**Review by harupy — APPROVED** [review]

**Comment by harupy** [comment]
LGTM!

**Pull request merged**